### PR TITLE
Lcstack3.1

### PR DESCRIPTION
**Pull request by zc0113**
提交栈20，155，232，三题

### CODE REVIEW

**Comment by ahhygx** [comment]
修改 要修改在原来的MR里，不是这个MR里

File: 155zc.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    int *array;
+} MinStack;
+
+/** initialize your data structure here. */
+#define EMPTYTOS (-1)
+MinStack* minStackCreate(int maxSize)
+{
+    MinStack *S;
+    S = malloc(sizeof(MinStack)*maxSize);
+    if(S == NULL)
```
Comment (ahhygx) [inline 155zc.c]: Short conditional statements may be written on one line if this enhances readability. You may use this only when the line is brief and the statement does not use the else clause.

https://google.github.io/styleguide/cppguide.html#Conditionals

File: 155zc.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    if(S->array == NULL)
+        return NULL;
+    S->topofstack = EMPTYTOS;
+    S->capacity = maxSize;
+    return S;
+}
+
+void minStackPush(MinStack* obj, int x)
+{
+    if(obj->topofstack == obj->capacity)
```
Comment (ahhygx) [inline 155zc.c]: In general, curly braces are not required for single-line statements, but they are allowed if you like them; conditional or loop statements with complex conditions or statements may be more readable with curly braces. Some projects require that an if must always have an accompanying brace.
https://google.github.io/styleguide/cppguide.html#Conditionals

File: 155zc.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return NULL;
+    else
+        obj->array[++obj->topofstack]=x;
+}
+
+void minStackPop(MinStack* obj)
+{
+    if(obj->topofstack == EMPTYTOS)
+        return NULL;
+    else
```
Comment (ahhygx) [inline 155zc.c]: 可以考虑把else去掉

File: 155zc.c (hunk 1 above)
Comment (ahhygx) [inline 155zc.c]: Note that in all cases you must have a space between the if and the open parenthesis. You must also have a space between the close parenthesis and the curly brace, if you're using one.

File: 155zc.c (hunk 2 above)
Comment (ahhygx) [inline 155zc.c]: Note that in all cases you must have a space between the if and the open parenthesis. You must also have a space between the close parenthesis and the curly brace, if you're using one.

File: 155zc.c [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+{
+    if( (obj->topofstack!=EMPTYTOS) )
+        return obj->array[obj->topofstack];
+    return 0;
+}
+
+int minStackGetMin(MinStack* obj)
+{
+    int min = obj->array[obj->topofstack];
+    int tmp = 0;
```
Comment (ahhygx) [inline 155zc.c]: define but not use

File: 155zc.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    for(int i=obj->topofstack;i>EMPTYTOS;i--)
+    {
```
Comment (ahhygx) [inline 155zc.c]: for () {
}

**Comment by zc0113** [comment]
是在原来的修改的，这个是新提交的分支，新的三题就是最后一个文件

File: 155zc.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+int minStackTop(MinStack* obj)
+{
+    if( (obj->topofstack!=EMPTYTOS) )
+        return obj->array[obj->topofstack];
+    return 0;
+}
+
+int minStackGetMin(MinStack* obj)
+{
+    int min = obj->array[obj->topofstack];
```
Comment (ahhygx) [inline 155zc.c]: Isn't it empty?

File: 155zc.c [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    int tmp = 0;
+    for(int i=obj->topofstack;i>EMPTYTOS;i--)
```
Comment (ahhygx) [inline 155zc.c]: =  > 两边的空格

File: 155zc.c [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        if( obj->array[i] < min)
+            min = obj->array[i];
+    }
+    return min;
+}
+
+void minStackFree(MinStack* obj)
+{
+    free(obj);
```
Comment (ahhygx) [inline 155zc.c]: if (obj)  free(obj);

Comment (ahhygx) [reply 155zc.c]: if obj is NULL , you will get crash!

File: 155zc.c [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    int topofstack;
+    int *array;
+} MinStack;
+
+/** initialize your data structure here. */
+#define EMPTYTOS (-1)
+MinStack* minStackCreate(int maxSize)
+{
+    MinStack *S;
+    S = malloc(sizeof(MinStack)*maxSize);
```
Comment (ahhygx) [inline 155zc.c]: why did you malloc so many stack ?

File: 20zc.c [hunk 10]
```diff
@@ -0,0 +1,24 @@
+bool isValid(char* s)
+{
+    if(s == "") return false;
```
Comment (ahhygx) [inline 20zc.c]: if 后面有空格

File: 20zc.c [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    int i=0,j=0;
```
Comment (ahhygx) [inline 20zc.c]: 逗号后面有1个空格

File: 20zc.c [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    char ch[10000];
```
Comment (ahhygx) [inline 20zc.c:5]: ch是干嘛的,用到了吗

File: 20zc.c [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    while(s[i] != '\0')
+    {
+        if(s[0] == ']' || s[0] =='}' || s[0] == ')') return false;
```
Comment (ahhygx) [inline 20zc.c]: 为何每次循环都要这个if ？

File: 20zc.c [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+        switch (s[i]){
+            case '{' : ch[j] = '{';break;
+            case '(' : ch[j] = '(';break;
+            case '[' : ch[j] = '[';break;
+            case ']' : if(j==0) return false; if(ch[j-1]=='[') j=j-2; else return false;break;
```
Comment (ahhygx) [inline 20zc.c]: 能写到多行吗，看不懂你写的啥，或者要花很久看你到底写的啥

File: 20zc.c [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+            case '}' : if(j==0) return false; if(ch[j-1]=='{') j=j-2; else return false;break;
+            case ')' : if(j==0) return false; if(ch[j-1]=='(') j=j-2; else return false;break;
+            }
+        i++;
+        j++;
+    }
+    if(j == 0)
```
Comment (ahhygx) [inline 20zc.c]: {}

**Comment by zc0113** [comment]
修改20zc.c  155zc.c

**Comment by zc0113** [comment]
修改232zc.c

File: 155zc.c (hunk 6 above)
Comment (ahhygx) [reply 155zc.c]: obj->topofstack 会等于-1吗

File: 155zc.c [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+int minStackTop(MinStack* obj){
+    if ((obj->topofstack!=EMPTYTOS))
+        return obj->array[obj->topofstack];
+    return 0;
+}
+
+int minStackGetMin(MinStack* obj){
+    int min = obj->array[obj->topofstack];
+    for (int i = obj->topofstack; i > EMPTYTOS; i--) {
+        if (obj->array[i] < min)
```
Comment (ahhygx) [inline 155zc.c]: 简单if 同一行

File: 20zc.c [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+                ch[j] = '(';
+                break;
+            }
+            case '[' : {
+                ch[j] = '[';
+                break;
+            }
+            case ']' : {
+                if (j == 0) return false;
+                if (ch[j-1] == '[')
```
Comment (ahhygx) [inline 20zc.c]: if else都存在都时候要加花括号

File: 20zc.c [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+                    j = j-2;
+                else
+                    return false;
+                break;
+            }
+        }
+        i++;
+        j++;
+    }
+    if (j == 0)
```
Comment (ahhygx) [inline 20zc.c]: 加花括号

File: 232zc.c [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+    return obj->arr[obj->top];
+}
+
+/** Returns whether the queue is empty. */
+bool myQueueEmpty(MyQueue* obj) {
+    return obj->top == -1 ? 1 : 0;
+}
+
+void myQueueFree(MyQueue* obj) {
+    obj->top = -1;
```
Comment (ahhygx) [inline 232zc.c:50]: 内存泄漏了， 一个malloc就要有对应都free

File: 496zc.c [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+{
+    int* a = (int*)malloc(sizeof(int) * findNumsSize);
+    memset(a,-1,findNumsSize * sizeof(int));
+    *returnSize = findNumsSize;
+    for(int i = 0;i < findNumsSize;i++)
+    {
+        int mark = 0;
+        for(int j = 0;j < numsSize;j++)
+        {
+            if(findNums[i] == nums[j])
```
Comment (ahhygx) [inline 496zc.c]: 简单if语句不需要两行

File: 496zc.c [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+                mark = 1;
+            if(findNums[i] < nums[j] && mark == 1)
```
Comment (ahhygx) [inline 496zc.c]: if 和（中间有空格

File: 682zc.c [hunk 22]
```diff
@@ -0,0 +1,45 @@
+int calPoints(char** ops, int opsSize)
+{
+    struct stack1
```
Comment (ahhygx) [inline 682zc.c:3]: stack1 -> Stack ？

File: 682zc.c [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+        int array[opsSize];
+        int pos;
+    };
+	int num = 0;
+	int i =0;
+    typedef struct stack1 stack;
+    stack* stack_1 = (stack*)malloc(sizeof(stack));
+    memset(stack_1->array,0,opsSize);
+    stack_1->pos = 0;
+    while(i<opsSize)
```
Comment (ahhygx) [inline 682zc.c:14]: while 和 （中间有空格

File: 682zc.c [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        switch(*(*(ops+i)))
+        {
```
Comment (ahhygx) [inline 682zc.c:17]: switch和 （中间有空格

File: 682zc.c [hunk 25]
```diff
@@ -0,0 +1,45 @@
...
+                i++;
+                break;
+            default:
+                stack_1->array[stack_1->pos] =atoi(*(ops+i));
+                stack_1->pos++;
+                i++;
+                break;
+        }
+    }
+    for(i = 1;i<stack_1->pos;i++)
```
Comment (ahhygx) [inline 682zc.c:40]: for 和 （中间有空格， ; 后面有空格

File: 682zc.c [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        stack_1->array[0]+=stack_1->array[i];
```
Comment (ahhygx) [inline 682zc.c:42]: += 两边有空格

**Comment by zc0113** [comment]
修改20/155

File: 682zc.c [hunk 27]
```diff
@@ -1,20 +1,17 @@
...
-    {
+    struct Stack {
         int array[opsSize];
         int pos;
     };
 	int num = 0;
-	int i =0;
-    typedef struct stack1 stack;
+	int i = 0;
+    typedef struct Stack stack;
```
Comment (ahhygx) [inline 682zc.c]: struct名字要用大写，不要用小写的struct

File: 844zc.c [hunk 28]
```diff
@@ -0,0 +1,40 @@
+int  stack(char *arry,char* arry2)
+{
+	int count = 0;
+	while (*arry != '\0')
+	{
+		if (*arry != '#')
```
Comment (ahhygx) [inline 844zc.c]: 这里是退4个空格不是8个空格

File: 844zc.c [hunk 29]
```diff
@@ -0,0 +1,40 @@
...
+		{
+			arry2[count] = *arry;
+			count++;
+		}
+		else if(count > 0)
+        {
```
Comment (ahhygx) [inline 844zc.c]: 这个是while循环里面的还外面是不是要和上面的if对齐

File: 844zc.c [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+            count--;
+        }else
+        {
+            count = 0;
+        }
+		arry++;
```
Comment (ahhygx) [inline 844zc.c:12]: 去掉4个空格

File: 844zc.c [hunk 31]
```diff
@@ -0,0 +1,40 @@
...
+	}
+	return count;
+}
+bool backspaceCompare(char* S, char* T)
+{
+	char *arry_s = (char *)malloc(201*sizeof(char));
```
Comment (ahhygx) [inline 844zc.c]: 为何这个要退后4格

File: 844zc.c [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+    char *arry_t = (char *)malloc(201*sizeof(char));
+	int count_s = stack(S,arry_s);
+    int count_t = stack(T,arry_t);
+	if (count_s != count_t)
```
Comment (ahhygx) [inline 844zc.c]: 为何这个要退后4格

File: 844zc.c [hunk 33]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        free(arry_s);
+        free(arry_t);
+        return false;
+    }
+	else
```
Comment (ahhygx) [inline 844zc.c]: 为何这个要退后4格

File: 844zc.c [hunk 34]
```diff
@@ -0,0 +1,40 @@
...
+        }
+		arry++;
+	}
+	return count;
+}
+bool backspaceCompare(char* S, char* T)
+{
+	char *arry_s = (char *)malloc(201*sizeof(char));
+    char *arry_t = (char *)malloc(201*sizeof(char));
+	int count_s = stack(S,arry_s);
```
Comment (ahhygx) [inline 844zc.c:19]: 为何这个要退后4格

File: 682zc.c (hunk 27 above)
Comment (zc0113) [reply 682zc.c]: 大写的报错，unknown

**Comment by zc0113** [comment]
修改844，496

**Comment by zc0113** [comment]
232 free内存

**Pull request closed without merging**